### ISSUE TITLE
Bug with float32

### ISSUE THREAD

**Issue by onanysunday** [golang/go#44180]
### Go Playground

<pre>
$ go version

</pre>

### Does this issue reproduce with the latest release?
Also get the error on 1.15.8 

### What operating system and processor architecture are you using (`go env`)?
windows/amd64

<details><summary>

```
package main

import (
	"fmt"
	"time"
)

var s float32
var i float32

func main() {

	t0 := time.Now()
	for i = 1; i <= 20000000; i++ {
		s += i + i
	}
	t1 := time.Now()
	fmt.Printf("run time = %v\n", t1.Sub(t0))
	fmt.Printf("s = %v\n", s)
}
```

**Comment by davecheney** [comment]
You have hit the limit of float32 precision, run this program and you'll see that `i` does not converge on 2e07
```
package main

import (
	"fmt"
	"time"
)

var s float32
var i float32

func main() {

	t0 := time.Now()
	for i = 1; i <= 20000000; i++ {
		s += i + i
		fmt.Println(i)
	}
	t1 := time.Now()
	fmt.Printf("run time = %v\n", t1.Sub(t0))
	fmt.Printf("s = %v\n", s)
}
```
Closing as this is not a bug in Go

Unlike many projects, the Go project does not use GitHub Issues for general discussion or asking questions. GitHub Issues are used for tracking bugs and [proposals](https://golang.org/s/proposal-process) only.

For asking questions, see:

* [Stack Overflow](https://stackoverflow.com/questions/tagged/go) with questions tagged "go"

* [The Go Forum](https://forum.golangbridge.org/), a web-based forum

* [Gophers Slack](https://gophers.slack.com), use the [invite app](https://invite.slack.golangbridge.org/) for access. The `#general` channel is a good starting point.

* [Go Community on Hashnode](https://hashnode.com/n/go) with questions and posts tagged with "go"

* [The golang-nuts mailing list](https://groups.google.com/d/forum/golang-nuts)

* **IRC** channel #go-nuts on Freenode

**Comment by onanysunday** [comment]
Wow, thanks Dave. I do not use floats in loops, but did it to try to get a relative speed comparison. Very astute!
Thanks. I just started Go and you've given me confidence in the language and the people involved.
Much thanks!




    On Monday, February 8, 2021, 11:13:44 PM EST, Dave Cheney <notifications@github.com> wrote:  
 
 


You have hit the limit of float32 precision, run this program and you'll see that i does not converge on 2e07
package main

import (
	"fmt"
	"time"
)

var s float32
var i float32

func main() {

	t0 := time.Now()
	for i = 1; i <= 20000000; i++ {
		s += i + i
		fmt.Println(i)
	}
	t1 := time.Now()
	fmt.Printf("run time = %v\n", t1.Sub(t0))
	fmt.Printf("s = %v\n", s)
}

Closing as this is not a bug in Go

Unlike many projects, the Go project does not use GitHub Issues for general discussion or asking questions. GitHub Issues are used for tracking bugs and proposals only.

For asking questions, see:
   
   -    
Stack Overflow with questions tagged "go"

   -    
The Go Forum, a web-based forum

   -    
Gophers Slack, use the invite app for access. The #general channel is a good starting point.

   -    
Go Community on Hashnode with questions and posts tagged with "go"

   -    
The golang-nuts mailing list

   -    
IRC channel #go-nuts on Freenode
